src/components/RecipesList.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import RecipeService from '../services/RecipeService';
+import { Link } from 'react-router-dom';
+
+const RecipesList = () => {
+  const [recipes, setRecipes] = useState([]);
+  const [currentRecipe, setCurrentRecipe] = useState(null);
+  const [currentIndex, setCurrentIndex] = useState(-1);
+  const [searchName, setSearchName] = useState("");
+
+  useEffect(() => {
+    retrieveRecipes();
+  }, []);
+
+  const onChangeSearchName = event => {
+    const searchName = event.target.value;
+    setSearchName(searchName);
+  };
+
+  const retrieveRecipes = () => {
+    RecipeService.getAll()
+      .then(response => {
+        setRecipes(response.data);
+        console.log(response.data);
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  };
+
+  const setActiveRecipe = (recipe, index) => {
+    setCurrentRecipe(recipe);
+    setCurrentIndex(index);
+  };
+
+  const findByName = () => {
+    RecipeService.findByName(searchName)
+      .then(response => {
+        setRecipes(response.data);
+        console.log(response.data);
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  };
+
+  return (
+    <div className="list row">
+      <div className="col-md-8">
+        <div className="input-group mb-3">
+          <input
+            type="text"
+            className="form-control"
+            placeholder="Search by name"
+            value={searchName}
+            onChange={onChangeSearchName}
+            data-testid="recipes-search-bar"
+          />
+          <div className="input-group-append">
+            <button
+              className="btn btn-outline-secondary"
+              type="button"
+              onClick={findByName}
+              data-testid="recipes-search-btn"
+            >Search
+            </button>
+          </div>
+        </div>
+      </div>
+      <div className="col-md-6">
+        <h4 data-testid="recipes-header">Recipes List</h4>
```

Comment:
Please let's be mindful of when to use different tags. Here we are using an h4 but we haven't used any h1, h2 or h3